Configure OIDC properties when necessary

Some combinations are failing because there are missing properties:

```
Caused by: java.lang.RuntimeException: Failed to start quarkus
Caused by: io.quarkus.runtime.configuration.ConfigurationException: 'auth-server-url' is not present. Both 'auth-server-url' and 'client-id' or alternatively 'public-key' must be configured when the quarkus-oidc extension is enabled
```

Therefore, when the extension quarkus-oidc is present, these two properties should be added.

--- a/README.md
+++ b/README.md
@@ -65,6 +65,10 @@ By default, the order of the combinations is random. In order to disable this co
 mvn clean test -Dts.random-sort-extensions=false
 ```
 
+## How to write custom properties by extension
+
+By convention, the test suite will try to lookup to a property using the name of the extension. For example, if the quarkus extension is called "quarkus-oidc", then the framework will try to load the custom properties at `src/test/resources/oidc.properties` and append the content into the combination application properties file.
+
 # How to contribute
 
 The build instructions are available in the [contribution guide](CONTRIBUTING.md).

--- a/src/main/java/quarkus/extensions/combinator/maven/MavenGenerator.java
+++ b/src/main/java/quarkus/extensions/combinator/maven/MavenGenerator.java
@@ -1,6 +1,7 @@
 package quarkus.extensions.combinator.maven;
 
 import java.io.File;
+import java.util.Optional;
 import java.util.Set;
 
 import quarkus.extensions.combinator.Configuration;
@@ -13,6 +14,8 @@ public class MavenGenerator extends MavenCommand {
     private static final String PROJECT_ARTIFACT_ID = "projectArtifactId";
     private static final String PROJECT_VERSION = "projectVersion";
     private static final String PLATFORM_ARTIFACT_ID = "platformArtifactId";
+    private static final String PROPERTIES_FORMAT = ".properties";
+    private static final String APPLICATION_PROPERTIES = "src/main/resources/application" + PROPERTIES_FORMAT;
 
     private static final String EXTENSIONS_PARAM = "extensions";
 
@@ -29,6 +32,7 @@ public class MavenGenerator extends MavenCommand {
 
         runMavenCommand(withQuarkusPlugin(), withProjectGroupId(), withProjectArtifactId(), withProjectVersion(),
                 withPlatformArtifactId(), withExtensions());
+        updateApplicationProperties();
         return new MavenProject(getArtifactId(), projectAsWorkingDirectory());
     }
 
@@ -56,12 +60,15 @@ public class MavenGenerator extends MavenCommand {
         return String.format(QUARKUS_PLUGIN, Configuration.QUARKUS_VERSION.get());
     }
 
-    public static final MavenGenerator withExtensions(Set<String> extensions) {
-        return new MavenGenerator(extensions);
-    }
+    private void updateApplicationProperties() {
+        File applicationProperties = new File(projectAsWorkingDirectory(), APPLICATION_PROPERTIES);
 
-    private static final String generateArtifactId(Set<String> extensions) {
-        return String.join("-", extensions);
+        for (String extension : extensions) {
+            Optional.ofNullable(getClass().getClassLoader().getResourceAsStream(extension + PROPERTIES_FORMAT))
+                    .ifPresent(customPropertiesByExtension -> {
+                        FileUtils.appendInputStreamIntoFile(customPropertiesByExtension, applicationProperties);
+                    });
+        }
     }
 
     private File projectAsWorkingDirectory() {
@@ -70,5 +77,13 @@ public class MavenGenerator extends MavenCommand {
 
     private static File targetAsWorkingDirectory() {
         return new File("target");
+    }
+
+    public static final MavenGenerator withExtensions(Set<String> extensions) {
+        return new MavenGenerator(extensions);
+    }
+
+    private static final String generateArtifactId(Set<String> extensions) {
+        return String.join("-", extensions);
     }
 }

--- a/src/main/java/quarkus/extensions/combinator/utils/FileUtils.java
+++ b/src/main/java/quarkus/extensions/combinator/utils/FileUtils.java
@@ -2,8 +2,13 @@ package quarkus.extensions.combinator.utils;
 
 import java.io.File;
 import java.io.IOException;
+import java.io.InputStream;
 import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
 import java.util.Arrays;
+import java.util.List;
+
+import org.apache.commons.io.IOUtils;
 
 public final class FileUtils {
 
@@ -35,7 +40,16 @@ public final class FileUtils {
         }
     }
 
-    public static void writeLine(File file, String line) {
+    public static void appendInputStreamIntoFile(InputStream is, File file) {
+        try {
+            List<String> lines = IOUtils.readLines(is, StandardCharsets.UTF_8);
+            org.apache.commons.io.FileUtils.writeLines(file, lines, APPEND);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void appendLineIntoFile(String line, File file) {
         try {
             org.apache.commons.io.FileUtils.writeLines(file, Arrays.asList(line), APPEND);
         } catch (IOException e) {

--- a/src/test/java/quarkus/extensions/combinator/extensions/RecordFailedScenariosTestQuarkusCombinationExtension.java
+++ b/src/test/java/quarkus/extensions/combinator/extensions/RecordFailedScenariosTestQuarkusCombinationExtension.java
@@ -19,7 +19,7 @@ public class RecordFailedScenariosTestQuarkusCombinationExtension implements Tes
 
     @Override
     public void testFailed(ExtensionContext context, Throwable cause) {
-        FileUtils.writeLine(new File(FAILED_FILE), context.getDisplayName());
+        FileUtils.appendLineIntoFile(context.getDisplayName(), new File(FAILED_FILE));
     }
 
 }

--- a/src/test/resources/oidc.properties
+++ b/src/test/resources/oidc.properties
@@ -1,0 +1,5 @@
+quarkus.oidc.tenant-enabled=false
+
+# HTTP Security Configuration
+quarkus.http.auth.permission.authenticated.paths=/*
+quarkus.http.auth.permission.authenticated.policy=permit

